jwst/photom/photom.py

```diff
@@ -795,35 +797,37 @@ def photom_io(self, tabdata, order=None):
 
             # Compute a 2-D grid of conversion factors, as a function of wavelength
             if isinstance(self.input, datamodels.MultiSlitModel):
-
+                slit = self.input.slits[self.slitnum]
                 # The NIRSpec fixed-slit primary slit needs special handling if
                 # it contains a point source
                 if self.exptype.upper() == 'NRS_FIXEDSLIT' and \
-                   self.input.slits[self.slitnum].name == self.input.meta.instrument.fixed_slit and \
-                   self.input.slits[self.slitnum].source_type.upper() == 'POINT':
+                   slit.name == self.input.meta.instrument.fixed_slit and \
+                   slit.source_type.upper() == 'POINT':
 
                     # First, compute 2D array of photom correction values using
                     # uncorrected wavelengths, which is appropriate for a uniform source
-                    scalar_conv = conversion
-                    conversion, no_cal = self.create_2d_conversion(self.input.slits[self.slitnum],
-                                                                   self.exptype, scalar_conv,
+                    conversion_2d_uniform, no_cal = self.create_2d_conversion(slit,
+                                                                   self.exptype, conversion_uniform,
                                                                    waves, relresps, order,
                                                                    use_wavecorr=False)
-                    slit.photom_uniform = conversion  # store the result
+                    slit.photom_uniform = conversion_2d_uniform  # store the result
 
                     # Now repeat the process using corrected wavelength values,
                     # which is appropriate for a point source. This is the version of
                     # the correction that will actually get applied to the data below.
-                    conversion, no_cal = self.create_2d_conversion(self.input.slits[self.slitnum],
-                                                                   self.exptype, scalar_conv,
+                    conversion, no_cal = self.create_2d_conversion(slit,
+                                                                   self.exptype, conversion,
                                                                    waves, relresps, order,
                                                                    use_wavecorr=True)
                     slit.photom_point = conversion  # store the result
 
                 else:
-                    conversion, no_cal = self.create_2d_conversion(self.input.slits[self.slitnum],
+                    conversion, no_cal = self.create_2d_conversion(slit,
                                                                    self.exptype, conversion,
                                                                    waves, relresps, order)
+
+                    slit.photom_uniform = conversion
```

Comment:
Now that I've done the historical research, I don't think this addition is needed. It's in the branch of the if/else that applies if the slit does NOT contain a point source, in which case we don't need to store either `photom_point` or `photom_uniform`. Just apply the computed uniform-source correction to the data, without storing it in the output.